src/sources/ljm/ljm_test.py

```
@@ -1,45 +1,179 @@
+    mock_ljm = MagicMock()
+    main_module.ljm = mock_ljm
+    
+    receiver = Receiver(main_module.CHANNEL)
+    
+    # Wait for receiver to connect by attempting non-blocking recv
+    start = time.time()
+    while time.time() - start < 1.0:  # 1 second timeout
+        try:
+            receiver.recv_message(timeout=0)
+            break
+        except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)